src/components/m-button/index.scss

```diff
@@ -91,6 +129,26 @@
 
   @include button--x-padding($x-pad);
 
+  .m-button--icon {
+    &:not(.m-button--reverse) {
+      @include button--icon-adjust($x-pad);
+    }
+
+    &.m-button--reverse {
+      @include button--icon-adjust--reverse($x-pad);
+    }
+  }
+
+  .m-button__icon {
+    &:not(.m-button__icon--reverse) {
```

Comment:
nesting shouldn't be necessary to overrule `.m-button__icon`, just put it aftwerwards